frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceTable.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Table } from '~/components/table';
+import { Identifier } from '~/types';
+import { nodeResourceColumns } from './const';
+import NodeResourceTableRow from './NodeResourceTableRow';
+import ManageNodeResourceModal from './ManageNodeResourceModal';
+
+type NodeResourceTableProps = {
+  identifiers: Identifier[];
+  onUpdate: (identifiers: Identifier[]) => void;
+  viewOnly?: boolean;
+};
+
+const NodeResourceTable: React.FC<NodeResourceTableProps> = ({
+  identifiers,
+  onUpdate,
+  viewOnly,
+}) => {
+  const [editIdentifier, setEditIdentifier] = React.useState<Identifier | undefined>();
```

Comment:
you can use `useGenericObjectState` here so that you can pass setPropValue to the form. so you dont need 
```tsx
  const handleFieldUpdate = (field: keyof Identifier, value: unknown) => {
    onUpdate({ ...identifier, [field]: value });
  };
```